package/src/package/mod.rs

```diff
@@ -1,19 +1,26 @@
 mod manifest;
+mod section;
 mod serialization;
 
+pub use self::manifest::{PackageExport, PackageManifest};
+pub use self::section::{InvalidSectionIdError, Section, SectionId};
+
 use alloc::{format, string::String, sync::Arc, vec::Vec};
 
 use miden_assembly_syntax::{Library, Report, ast::QualifiedProcedureName};
 use miden_core::{Program, Word};
 
-pub use self::manifest::{PackageExport, PackageManifest};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
 use crate::MastArtifact;
 
 // PACKAGE
 // ================================================================================================
 
 /// A package containing a [Program]/[Library], and a manifest (exports and dependencies).
 #[derive(Debug, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct Package {
     /// Name of the package
     pub name: String,
```

Comment:
Should we add other things here like `description` and `version`? We currently have these in [AccountComponentMetadata](https://github.com/0xMiden/miden-base/blob/next/crates/miden-objects/src/account/component/template/mod.rs#L150), and it would seem a bit weird to have the name here and then description/version in a custom section.

We could also duplicate the name in both places, but that also seems a bit weird.

Reply:
Makes sense to me to have those as optional fields, definitely